eodhp_utils/aws/egress_classifier.py

```
@@ -0,0 +1,174 @@
+    fallback_file: str
+
+    def __init__(
+        self,
+        current_region: str = None,
+        url: str = "https://ip-ranges.amazonaws.com/ip-ranges.json",
+        cache_file: str = "/var/cache/eodhp/eodhp-utils/ip-ranges.json",
+    ):
+        if current_region is None:
+            # This uses AWS_DEFAULT_REGION from the environment.
+            #
+            # Fetching http://169.254.169.254/latest/meta-data/placement/availability-zone is an
+            # alternative if this proves inadequate.
+            current_region = boto3.session.Session().region_name
+
+        self.aws_ip_data_url = url
+        self.current_region = current_region
+        self.fallback_file = cache_file
+
+        self._build_tree()
+
+    def _build_tree(self):
+        ip_data = (
+            self._get_ip_data_from_aws()
+            or self._get_cached_ip_data()
+            or self._get_bundled_ip_data()
+        )
+
+        self.current_tree, self.aws_tree = self.build_trees(ip_data, self.current_region)
+        self.load_time = time.time()
+
+    def _get_ip_data_from_aws(self) -> Optional[dict]:
+        logger.debug("Fetching AWS IP address file from %s", self.aws_ip_data_url)
+
+        try:
+            response = requests.get(self.aws_ip_data_url)
+            response.raise_for_status()
+
+            logger.info("Fetched AWS IP address file from %s", self.aws_ip_data_url)
+            result_data = response.json()
+            assert isinstance(result_data, dict)
+
+            self._write_cache_file(response.content)
+
+            return result_data
+        except requests.RequestException:
+            logger.exception("Failed to fetch AWS IP address file at %s", self.aws_ip_data_url)
+            return None
+
+    def _get_cached_ip_data(self) -> Optional[dict]:
+        try:
+            with open(self.fallback_file, "r") as f:
+                ip_data = json.load(f)
+                return ip_data if isinstance(ip_data, dict) else None
```

If the fallback mechanism fails, Do you think we should raise a runtime error?